### PR TITLE
add example of resolvers to basic schema example

### LINKED ISSUE #1595: Add a complete example in schema basic documentation
This came up from discord, in the Schema Basic docs: https://strawberry.rocks/docs/general/schema-basics we have two types that reference each other but we don't show how to create resolvers for that. This is not beginners friendly

The easiest solution could be to clean up and copy this gist (https://gist.github.com/patrick91/362b4de2a341c879a53f8c2ae8588b7e) at the end of the page

### PR DESCRIPTION
**Pull request by tsmith023**
## Description

Added an example for writing resolvers to populate fields within types on the Schema Basics docs page.

It follows the flow of https://gist.github.com/patrick91/362b4de2a341c879a53f8c2ae8588b7e but with the data conforming with the rest of the examples on the Schema Basics page.

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Documentation

## Issues Fixed or Closed by This PR

* Closes #1595 

## Checklist

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the CONTRIBUTING document.
- [ ] I have added tests to cover my changes.
- [ ] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Comment by tsmith023** [comment]
@patrick91, Github won't let me add you as a reviewer for this PR and I'm unsure as to why :smile:

**Comment by DoctorJohn** [comment]
Thanks for the PR @tsmith023. Your changes to the docs look good to me. I requested a review from Patrick for you, not sure why GIthub won't let you do that :)

File: docs/general/schema-basics.md [hunk 1]
```diff
@@ -138,6 +138,60 @@ class Author:
   books: typing.List[Book]
 ```
 
+## Basic Resolvers
+
+In the above schema, a `Book` has an `author` field and an `Author` has a `books`
+field yet we do not know how our data can be mapped to fulfil the structure of
```
Comment (patrick91) [inline docs/general/schema-basics.md:144]:
```suggestion
field, yet we do not know how our data can be mapped to fulfil the structure of
```

File: docs/general/schema-basics.md [hunk 2]
```diff
@@ -138,6 +138,60 @@ class Author:
...
+the promised schema.
+
+To achieve this, we introduce the concept of the _resolver_ that provides some
```
Comment (patrick91) [inline docs/general/schema-basics.md:147]:
```suggestion
To achieve this, we introduce the concept of the [_resolver_](../types/resolvers.md) that provides some
```

File: docs/general/schema-basics.md [hunk 3]
```diff
@@ -138,6 +138,60 @@ class Author:
   books: typing.List[Book]
 ```
 
+## Basic Resolvers
```
Comment (patrick91) [inline docs/general/schema-basics.md:141]:
```suggestion
## Providing data to fields
```

File: docs/general/schema-basics.md [hunk 4]
```diff
@@ -138,6 +138,60 @@ class Author:
...
+
+In the above schema, a `Book` has an `author` field and an `Author` has a `books`
+field yet we do not know how our data can be mapped to fulfil the structure of
+the promised schema.
+
+To achieve this, we introduce the concept of the _resolver_ that provides some
+data to a field through a function.
+
+Continuing with this example of books and authors, basic resolvers can be defined
```
Comment (patrick91) [inline docs/general/schema-basics.md:150]:
```suggestion
Continuing with this example of books and authors, resolvers can be defined
```

File: docs/general/schema-basics.md [hunk 5]
```diff
@@ -138,6 +138,60 @@ class Author:
...
+These functions provide the `strawberry.field` with the ability to render data
+to the GraphQL query upon request and are the backbone of all GraphQL APIs.
+
+This example is trivial since the resolved data is entirely static. However,
+when building more complex APIs, these resolvers can be written to map data
+from databases, e.g. making SQL queries using SQLAlchemy, and other APIs,
+e.g. making HTTP requests using aiohttp.
+
+For more information and detail on the different ways to write resolvers,
+see the resolvers section.
```
Comment (patrick91) [inline docs/general/schema-basics.md:193]:
```suggestion
see the [resolvers section](../types/resolvers.md).
```

**Review by patrick91 — APPROVED** [review]
Looks pretty good, I've added a few small changes 😊

**Pull request merged**